### DISCUSSION TITLE
Issue with User Re-Authentication Using Google OAuth in Fief

### DISCUSSION THREAD

**Discussion by bohachu** [fief-dev/fief#314] [Q&A]
Hello Fief Community,

I'm experiencing a challenge with the user re-authentication flow when integrating Google OAuth 2.0 with Fief's authentication system. Here's a detailed description of the issue:

**Initial Authentication:** Users successfully authenticate with Google OAuth on their first attempt and are directed to Fief's registration page. Post-registration, they are able to log in to my service without any issues.

**Re-authentication Concern:** The issue arises when the same users try to re-authenticate using Google OAuth. Instead of being logged in, they are redirected to the registration page again. When they try to register, an error message indicates that the account already exists.

Upon inspecting the database, I noticed something that might be contributing to this issue. In the `fief_oauth_accounts` table, there is an entry for users who registered via Google OAuth, but the `user_id` field is `NULL`. This seems to suggest that the system is unable to associate the returning user with an existing account, leading to the redirection to the registration page during subsequent login attempts.

I'm wondering if there might be a configuration step I missed or if there's a specific setting in Fief that needs to be adjusted to handle this scenario more appropriately. Any insights or suggestions to resolve this behavior would be greatly appreciated.

Thank you for your assistance!

**Comment by fief-bailiff** [comment]
Hail, @bohachu 👋 Welcome to Fief's kingdom!

Our team will get back to you very soon to help.

In the meantime, take a minute to star our repository ⭐️

![star-fief](https://user-images.githubusercontent.com/1144727/192523552-dc2c31fa-ae94-42fa-9ab4-7d0ba51bb7b0.png)

Farewell!

**Comment by BlackIsBlack** [comment]
Bump. Experiencing the exact same issue. Is this a missed config step? This is currently limiting me in a production environment. Please help

**Comment by frankie567** [comment]
Fixed in [0.28.8](https://github.com/fief-dev/fief/releases/tag/v0.28.8). Thank you @BlackIsBlack for bumping this, I'm sorry to not have catched the original post sooner.
